altimeter_ms5607_spi/altimeter_ms5607_spi.cc

```diff
@@ -0,0 +1,142 @@
+#include "altimeter_ms5607_spi.h"
+
+#include <cmath>
+
+AltimeterMs5607Spi::AltimeterMs5607Spi(SPI_HandleTypeDef *hspi, GPIO_TypeDef *csPort, uint16_t csPin, GPIO_TypeDef *misoPort, uint16_t misoPin, double seaLevelPressure)
+    : _hspi(hspi), _csPort(csPort), _csPin(csPin), _misoPort(misoPort), _misoPin(misoPin), _seaLevelPressure(seaLevelPressure) {}
+
+bool AltimeterMs5607Spi::Reset() {
+    bool opStatus = true;
+
+    HAL_GPIO_WritePin(_csPort, _csPin, GPIO_PIN_RESET);
+    // reset, opcode 0x1E
+    if (HAL_SPI_Transmit(_hspi, (uint8_t *)(const uint8_t[]){0x1E}, 1, SERIAL_TIMEOUT) != HAL_OK) opStatus = false;
+    // poll for reset complete (approx 2.8ms)
+    GPIO_PinState readStatus = GPIO_PIN_RESET;
+    while (readStatus == GPIO_PIN_RESET) {
```

Comment:
this is kinda paranoid, but any potential for infinite loop is scary. Can we do a max tries before this will give up and return a status indicating failure?